doc/tutorials/workshop/content/exercises/13-complex-yaml.md

```diff
@@ -0,0 +1,103 @@
+
+```
+$ oc get configmap config -n openshift-kube-apiserver -oyaml
+apiVersion: v1
+data:
+  config.yaml: '{"admission":{"pluginConfig":{"PodSecurity":{"configuration":{"apiVersion":"pod-security.admission.conf
+ig.k8s.io/v1","defaults":{"audit":"restricted","audit-version":"latest","enforce":"privileged","enforce-version":"lates
+t","warn":"restricted","warn-version":"latest"},"exemptions":{"usernames":["system:serviceaccount:openshift-infra:build
+-controller"]},"kind":"PodSecurityConfiguration"}},"network.openshift.io/ExternalIPRanger":{"configuration":{"allowIngr
+essIP":false,"apiVersion":"network.openshift.io/v1","externalIPNetworkCIDRs":null,"kind":"ExternalIPRangerAdmissionConf
+ig"},"location":""},"network.openshift.io/RestrictedEndpointsAdmission":{"configuration":{"apiVersion":"network.openshi
+ft.io/v1","kind":"RestrictedEndpointsAdmissionConfig","restrictedCIDRs":["10.128.0.0/14","172.30.0.0/16"]}}}},"apiServe
+rArguments":{"allow-privileged":["true"],"anonymous-auth":["true"],"api-audiences":["https://kubernetes.default.svc"],"
+audit-log-format":["json"],"audit-log-maxbackup":["10"],"audit-log-maxsize":["200"],"audit-log-path":["/var/log/kube-ap
+iserver/audit.log"],"audit-policy-file":["/etc/kubernetes/static-pod-resources/configmaps/kube-apiserver-audit-policies
+...
+kind: ConfigMap
+metadata:
+  creationTimestamp: "2023-10-06T12:59:31Z"
+  name: config
+  namespace: openshift-kube-apiserver
+  resourceVersion: "21469"
+  uid: fc192e71-282a-4af6-9492-87324ea83410
+```
+
+To check the value of `audit-log-maxbackup`, which is in the `config.yaml` key,
+we need to use the `jq` query to select the value and "extract" it for us.
+
+Let's create a machine config that has a nested yaml value:
+```
+$ cat << EOF | oc create -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-nested-compliance-configmap
+  namespace: openshift
+data:
+  my-config.yaml: '{"foo": "bar", "nested-key": "nested-compliant"}'
+EOF
+```
+
+Now let's write a rule to check whether the value of `nested-key` is `nested-compliant`.
+
+```
+$ ./utils/add_kubernetes_rule.py create platform \
+    --rule check_nested_yaml \
+    --name my-nested-compliance-configmap --namespace openshift --type configmap \
+    --title "Check value of nested-key in my-config-yaml" \
+    --description "It is important that the nested configmap has a nested-key with value nested-compliant." \
+    --match-entity "at least one" \
+    --match "nested-compliant" \
+    --yamlpath ".nested-key" \
+    --jqfilter '.data."my-config.yaml"'
+```
+
+And then you can test the rule with:
+```
+$ ./utils/add_platform_rule.py cluster-test --rule check_nested_yaml
+```
+
+## 2. Filtering the data to have simpler yamlpaths
+
+This is a generalization of the first use case, `jq` filters can be used to filter and select the data
+fetched by the operator.
+
+When the resource being checked is extensive or complex, `jq` is prety powerful and using it to filter and trim down
```

Comment:
Technically, we wouldn't  have any other way to do extensive or complex checking without jq.

If so, we could make this sentence more assertive:

When the resource being checked is an extensive or complex, `jq` is invaluable for simplifying the data before it's evaluated with `yamlpath`.